### PR TITLE
Adds Community Nurse's Log; improved Nursing Care section UI

### LINKED ISSUE #8488: Log update : Round Type (Community Nurse's Log)
**Is your feature request related to a problem? Please describe.**
CARE is being adopted by palliative care organisations. 

We need a new round type specifically for community nurses.

**Describe the solution you'd like**

Have the following fields in the form in the given order

- [ ] Measured at
- [ ] Round type
- [ ] category
- [ ] Symptoms
- [ ] Physical examination info
- [ ] Other details
_____________________________ the below section comprises of new fields to be added
- ROUTINE 
- [ ] Bowel (dropdown with options- No difficulty, Constipation, Diarrhoea)
- [ ] Bladder 
- Bladder Drainage (dropdown with options- Normal, Condom Catheter, Diaper, Intermittent catheter, Continuous indwelling catheter, Continuous suprapubic catheter, Urostomy)
- Bladder Issues (dropdown with options- No issues, Incontinence , Retention, Hesitancy)
- Whether the patient experiences Dysuria? Yes/No
- Frequency of urination (dropdown with options- Increased frequency, Normal, Decreased frequency)
- [ ] Sleep (dropdown with options- Excessive, Satisfactory, Unsatisfactory, No sleep )
- [ ] Nutrition
- Nutrition route (dropdown with options- Oral, Ryle’s tube, Gastrostomy/jejunostomy, PEG, Parenteral Tubing (fluid), Parenteral Tubing(TPN))
- If Oral, Issues (dropdown with options- No issue, Dysphagia, Odynophagia)
- Appetite (dropdown with options- Increased, Satisfactory, Reduced, No taste for food, Cannot assess)
_____________________________
- VITALS
- [ ] BP - systolic/diastolic
- [ ] Pulse
- [ ] RBS (mg/dl) (same as Blood sugar level in detailed log update)
____________________________
- PRESCRIPTION
keep the prescription section as is
__________________________
NURSING CARE
keep entire nursing care section as is
_________________________
- [ ] Action review after
- [ ] Review After

**Describe alternatives you've considered**

**Additional context**

### LINKED ISSUE #8521: Improving Nursing Care UI in Log Update
**Is your feature request related to a problem? Please describe.**
Currently the nursing care UI is not optimal, it should ideally be a multi-select of procedures followed by a table with remarks column.

cc: @bodhish 

### Current

image

#### When empty

image

#### When selected

image

### LINKED ISSUE #8532: Hide sections in log update details page if no data present
**Is your feature request related to a problem? Please describe.**
Currently the view log update details page shows all data eve if most of them maybe blank.

**Describe the solution you'd like**
Hide sections and details in log update details page if no related data are present

image

### LINKED ISSUE #8534: Patient category not shown in log update details page
> Originally posted by @nihal467 in https://github.com/ohcnetwork/care_fe/pull/8522#issuecomment-2348897053

**Describe the bug**
Patient category not shown in log update details page

**To Reproduce**

1. File a detailed log update with a category
2. Click on View Details page of the filed log update.

**Screenshots**
image

### LINKED ISSUE #8557: Nursing Information  : visualising data better
**Is your feature request related to a problem? Please describe.**
The community nurses log update introduces the "Routine section". The details entered therein must be listed in the "Nursing Tab" in patient dashboard

**Describe the solution you'd like**
In Nursing tab, in patient dashboard, make the following changes:

- [x] Change title from "Nursing Analysis" to "Nursing Information"
- [x] Add a section for "Routine" and add a table in the following format
Screenshot 2024-09-17 at 4 01 20 PM
- [x] Add another section below it for "Nursing Care". 
- [ ] Show nursing care also in a similar table. Add columns for dates and rows for nursing care given. Add a tick mark in the cell when a nursing care procedure is done on a said date. Notes for the same may be marked with a "note" icon and visible to user upon clicking it. (design similar to Medicine administration record)

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by rithviknishad**
## Required Backend Changes

- https://github.com/ohcnetwork/care/pull/2448

## Proposed Changes

- Fixes #8488
- Fixes #8521
- Fixes #8532
- Fixes #8534
- Fixes #8557

### 1. Improved Nursing Care UX

Use the multi-select menu to select all of the relevant nursing care procedures to be filed, and then add remarks for each of the nursing care procedures.

image

image

### 2. Adds support for filing Community Nurse's log update

image

### 3. Hide's details from Log Update details page if section/data is not filled.

image

### 4. Updated Nursing Information tab to show Routine information

image

@ohcnetwork/care-fe-code-reviewers

## Merge Checklist

- [ ] Add specs that demonstrate bug / test a new feature.
- [ ] Update [product documentation](https://docs.ohc.network).
- [ ] Ensure that UI text is kept in I18n files.
- [ ] Prep screenshot or demo video for changelog entry, and attach it to issue.
- [ ] Request for Peer Reviews
- [ ] Completion of QA

### CODE REVIEW

**Comment by nihal467** [comment]
- [x] creation
- [x] updation
- [x] edit
- [x] view in charts
- [x] view in details page
- [x] events log

**Comment by nihal467** [comment]
**Bug**

image

- [x] the dropdowns are not properly responsive

image
image


- [x] in the view details page, the neurological monitor table are shown, which are not being recorded, so show only relevant and recorded details in this page
- [x] similarly hide the pain scale and pressure sore too

image
image

- [x] when you click update log button, it is been redirected to a irrelevant page, the user should be able to view the form directly

image

- [x] nursing care without remarks are hidden in the view details page

image

- [x] fix the typo error for the log update notification and post submission redirection

image

- [x] the patient category details in missing in this page

- [ ] the newly added fields have no reflection across the patient consultation page, over any cards, 

Enhancement

- [ ] only show oral issue, if the user select nutrition route as oral, otherwise hide it

**Comment by nihal467** [comment]
LGTM , conditionally approving this PR as its priority, the data reflection card needs to be followed up with @aparnacoronasafe , and create a new issue for that as well @rithviknishad @khavinshankar

**Comment by nihal467** [comment]
image

- [ ] @rithviknishad as an enhancement, lock the first row of the field name

**Comment by nihal467** [comment]
LGTM

**Review by nihal467 — APPROVED** [review]

File: src/Components/Facility/ConsultationDetails/ConsultationNursingTab.tsx [hunk 1]
```diff
@@ -1,18 +1,226 @@
...
+              { title: "Bladder" } as const,
+              { subField: true, field: "bladder_drainage" } as const,
+              { subField: true, field: "bladder_issue" } as const,
+              { subField: true, field: "is_experiencing_dysuria" } as const,
+              { subField: true, field: "urination_frequency" } as const,
+              { title: "Nutrition" } as const,
+              { subField: true, field: "nutrition_route" } as const,
+              { subField: true, field: "oral_issue" } as const,
+              { subField: true, field: "appetite" } as const,
+            ].map((row) => (
```
Comment (khavinshankar) [inline src/Components/Facility/ConsultationDetails/ConsultationNursingTab.tsx]: can we separate it out as a const as it's not dynamic

**Review by khavinshankar — COMMENTED** [review]
LGTM, just a minor refactor

**Review by khavinshankar — APPROVED** [review]

**Pull request merged**